### PR TITLE
Converts join-clause-part into a slice of join-clause-parts

### LINKED ISSUE #130: parser: JoinClause.JoinClausePart must be slice
In the AST, according to https://www.sqlite.org/syntax/join-clause.html , JoinClause.JoinClausePart must be a slice.
With the current state, the AST doesn't allow to represent statements like `SELECT * FROM a JOIN b JOIN c`

### PR DESCRIPTION
**Pull request by SUMUKHA-PK**
Fixes #130 

Bug fixed, test added to verify the change.

### CODE REVIEW

File: internal/parser/parser_test.go [hunk 1]
```diff
@@ -1799,8 +1801,8 @@ func TestSingleStatementParse(t *testing.T) {
 			},
 		},
 		{
-			"DELETE with basic with clause, select stmt with FROM with joinclause's ON and basic cte-table-name",
-			"WITH myTable AS (SELECT * FROM myTable1,myTable2 ON myExpr) DELETE FROM myTable",
+			"DELETE with basic with clause, select stmt with FROM with joinclause with multiple join-clause-parts, join constraint and basic cte-table-name",
```
Comment (TimSatke) [inline internal/parser/parser_test.go:1804]: don't change existing tests, add new ones

Comment (SUMUKHA-PK) [reply internal/parser/parser_test.go:1804]: The changes to the old tests are to accomodate for the slice than the single element. 
The test added is indeed a new one, old tests arent removed :D

Comment (SUMUKHA-PK) [reply internal/parser/parser_test.go:1804]: The changes in the old tests are to accomodate for the change of the DS to a slice.
The test indeed is a new one, no old tests were removed :D

File: internal/parser/parser_test.go [hunk 2]
```diff
@@ -1799,8 +1801,8 @@ func TestSingleStatementParse(t *testing.T) {
...
+			"WITH myTable AS (SELECT * FROM myTable1,myTable2 JOIN myTable3) DELETE FROM myTable",
```
Comment (TimSatke) [inline internal/parser/parser_test.go:1805]: what about this?

Comment (TimSatke) [reply internal/parser/parser_test.go:1805]: Diff tricked us. This is a new test, old test is below as addition.

**Pull request merged**